run_cert.py

```diff
@@ -0,0 +1,333 @@
+DOMAINS_FILE = "/etc/letsencrypt/current_domains"
+
+DEFAULT_CERTBOT_ARGS = [
+    "certbot",
+    "certonly",
+    "--server", os.environ.get(ENV_LETSENCRYPT_SERVER_URL, DEFAULT_LETSENCRYPT_URL),
+    "--email", os.environ.get(ENV_LETSENCRYPT_EMAIL),
+    "--agree-tos",
+    "--noninteractive",
+    "--rsa-key-size", os.environ.get(ENV_RSA_KEY_SIZE, "4096"),
+    "--expand"
+]
+
+CERTBOT_ARGS_HTTP = [
+    "--standalone",
+    "--no-redirect",
+    "--preferred-challenges",
+    "http-01"
+]
+
+CERTBOT_ARGS_DNS = [
+    "--preferred-challenges",
+    "dns-01"
+]
+
+CERTBOT_ARGS_DNS_GCLOUD = [
+    "--dns-google",
+    "--dns-google-credentials", os.environ.get(ENV_GOOGLE_CREDENTIALS, DEFAULT_GCE_CREDENTIALS),
+    "--dns-google-propagation-seconds", os.environ.get(ENV_DNS_PROPAGATION_TIMEOUT, "120")
+]
+
+CERTBOT_ARGS_DNS_ROUTE53 = [
+    "--dns-route53",
+    "--dns-route53-propagation-seconds", os.environ.get(ENV_DNS_PROPAGATION_TIMEOUT, "120")
+]
+
+
+def get_marathon_url():
+    """Retrieves the marathon base url to use from an environment variable"""
+    return os.environ.get(ENV_MARATHON_URL, DEFAULT_MARATHON_URL)
+
+
+def get_letsencrypt_url():
+    """Retrieves the LetsEncrypt Server URL"""
+    return os.environ.get(ENV_LETSENCRYPT_SERVER_URL, DEFAULT_LETSENCRYPT_URL)
+
+
+def get_marathon_app(app_id):
+    """Retrieve app definition for marathon-lb app"""
+    response = requests.get("%(marathon_url)s/v2/apps/%(app_id)s"
+                            % dict(marathon_url=get_marathon_url(), app_id=app_id), verify=False)
+    if not response.ok:
+        raise Exception("Could not get app details from marathon")
+    return response.json()
+
+
+def read_domains_from_last_time():
+    """Return list of domains used (last time this script was run) from file or empty string if file does not exist"""
+    if os.path.exists(DOMAINS_FILE):
+        with open(DOMAINS_FILE) as domains_file:
+            return domains_file.read()
+    else:
+        return ""
+
+
+def write_domains_to_file(domains):
+    """Store list of domains in file to retrieve on next run"""
+    with open(DOMAINS_FILE, "w+") as domains_file:
+        domains_file.write(domains)
+
+
+def find_newest_dir(domain_name):
+    dirs = {}
+    # Check if the certificate has been recreated with a new domain list
+    for x in os.listdir(CERTIFICATES_DIR):
+        if x.startswith(domain_name):
+            # Dict all the directories and their creation times in the CERTIFICATES_DIR that start with the domain_name
+            xpath = f"{CERTIFICATES_DIR}/{x}"
+            dirs[xpath] = os.path.getctime(xpath)
+    # Return the newest directory in the dict
+    list_dir = sorted(dirs.items(), key=operator.itemgetter(1))
+    return list_dir[-1][0]
+
+
+def write_combined_cert_to_file(domain_name):
+    """Create the combined cert from the full chain and private key files"""
+    if domain_name.startswith("*"):
+        domain_name = domain_name.replace("*.", "")
```

Comment:
These same 2 lines above are repeated later in the file so could be reused in a function.